### PR TITLE
fix(openapi-generator): fix template return type

### LINKED ISSUE #183: OpenAPI Generator not generating correct model for arrays
**Describe the bug**
OpenAPI Generator is not generating correct model for some api responses (example: arrays)

**To Reproduce**
1. Generate api with swagger.json which included an api response of type `Array<SomeModel>`
2. Inspeact the generated API
3. The api in question is generated as AxiosResponse<SomeModel> instead of AxiosResponse<Array<SomeModel>>

**Expected behavior**
Should generate correct models.

**Additional context**
I changes the template from typescript-axios to typescript-fetch and it generates correctly

### PR DESCRIPTION
**Pull request by kherP**
fix #183

### CODE REVIEW

**Review by omarVodiaK — APPROVED** [review]

File: src/openapiGenerator/mockGenerator/formatResponse.ts [hunk 1]
```diff
@@ -35,10 +35,38 @@ export const extractResponses = (obj: OpenAPI.Document): ResponsesType => {
                 } else if (response.content && response.content[APPLICATION_JSON]?.schema) {
                     mock = response.content[APPLICATION_JSON].schema;
                 }
-                mock = typeof(mock) === "string" || isExample ? mock : generateData("", mock, modelSchemas);
+                mock = typeof (mock) === "string" || isExample ? mock : generateData("", mock, modelSchemas);
```
Comment (mario-subo) [inline src/openapiGenerator/mockGenerator/formatResponse.ts:38]: this should be `typeof(mock)` or `typeof mock`

image

File: src/openapiGenerator/mockGenerator/formatResponse.ts [hunk 2]
```diff
@@ -35,10 +35,38 @@ export const extractResponses = (obj: OpenAPI.Document): ResponsesType => {
...
+    const pathArr = path.split("/").slice(1);
+    pathArr.forEach((item: string, index: number) => {
+        let newItem = item.replace(/[^\w\s]/gi, "");
+        if (index > 0) {
+            newItem = sentenceCase(newItem)
+        }
+        newString += newItem;
+    })
+    return newString + sentenceCase(method);
+}
```
Comment (mario-subo) [inline src/openapiGenerator/mockGenerator/formatResponse.ts:63]: missing types for `newString`, `pathArr` and `newItem`

**Review by mario-subo — CHANGES_REQUESTED** [review]

File: src/openapiGenerator/mockGenerator/formatResponse.ts (hunk 2 above)
Comment (kherP) [reply src/openapiGenerator/mockGenerator/formatResponse.ts]: added

File: src/openapiGenerator/mockGenerator/formatResponse.ts (hunk 1 above)
Comment (kherP) [reply src/openapiGenerator/mockGenerator/formatResponse.ts]: changed to typeof operand

**Review by yousifalraheem — APPROVED** [review]

**Review by mario-subo — APPROVED** [review]

**Pull request merged**